### PR TITLE
Add Maven registry metadata to Java/Kotlin plugins

### PR DESCRIPTION
**Pull request by pkwarren**

### CODE REVIEW

File: plugins/bufbuild/connect-kotlin/v0.1.1/buf.plugin.yaml [hunk 1]
```diff
@@ -9,3 +9,16 @@ output_languages:
   - kotlin
 spdx_license_id: Apache-2.0
 license_url: https://github.com/bufbuild/connect-kotlin/blob/v0.1.1/LICENSE
+registry:
+  maven:
+    compiler:
+      kotlin:
+        version: 1.8.10
```
Comment (pkwarren) [inline plugins/bufbuild/connect-kotlin/v0.1.1/buf.plugin.yaml:16]: I'd like to validate these in the tests.

File: plugins/bufbuild/connect-kotlin/v0.1.1/buf.plugin.yaml [hunk 2]
```diff
@@ -9,3 +9,16 @@ output_languages:
...
+    deps:
+      - build.buf:connect-kotlin:0.1.1
+      - build.buf:connect-kotlin-google-java-ext:0.1.1
```
Comment (pkwarren) [inline plugins/bufbuild/connect-kotlin/v0.1.1/buf.plugin.yaml:19]: I also think we should validate all deps can be resolved from Maven central.

File: plugins/protocolbuffers/kotlin/v21.10/buf.plugin.yaml [hunk 3]
```diff
@@ -7,3 +7,17 @@ output_languages:
...
 spdx_license_id: BSD-3-Clause
 license_url: https://github.com/protocolbuffers/protobuf/blob/v21.10/LICENSE
+registry:
+  maven:
+    compiler:
+      kotlin:
+        version: 1.8.10
+    deps:
+      - com.google.protobuf:protobuf-kotlin:3.21.10
+      - org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.8.10
```
Comment (pkwarren) [inline plugins/protocolbuffers/kotlin/v21.10/buf.plugin.yaml:17]: Added jdk8 dependency to full runtime, and just normal stdlib to lite runtime. Does that sound reasonable @buildbreaker? I figure anyone using these dependencies can always pull in additional support for newer features.

**Review by stefanvanburen — APPROVED** [review]
lgtm, although may want to wait on @buildbreaker's thoughts

**Pull request merged**